bxin/ptg2hex/hex_diagnostics.ipynb

```diff
@@ -0,0 +1,607 @@
+    "print('staring with: hex state', salobj.State(state.summaryState), pd.to_datetime(state.private_sndStamp, unit='s'))\n",
+    "if state.summaryState == 2:\n",
+    "    await salobj.set_summary_state(hex, salobj.State.DISABLED) #disable hex\n",
+    "    await asyncio.sleep(3)\n",
+    "    state = await hex.evt_summaryState.aget(timeout=5)\n",
+    "    print('hex state', salobj.State(state.summaryState), pd.to_datetime(state.private_sndStamp, unit='s'))\n",
+    "\n",
+    "await salobj.set_summary_state(hex, salobj.State.ENABLED, settingsToApply=\"default\") #enable hex\n",
+    "state = await hex.evt_summaryState.aget(timeout=5)\n",
+    "print('hex state', salobj.State(state.summaryState), pd.to_datetime(state.private_sndStamp, unit='s'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#may need to wait a few seconds before event shows up in EFD\n",
+    "await asyncio.sleep(5.)\n",
+    "from lsst_efd_client import EfdClient\n",
+    "\n",
+    "client = EfdClient('ncsa_teststand_efd')\n",
+    "csc_index = 1\n",
+    "end = Time(datetime.now())\n",
+    "start = end - timedelta(hours=1)\n",
+    "dfe = await client.select_time_series('lsst.sal.MTHexapod.logevent_summaryState', '*', start.tai, end.tai, csc_index)\n",
+    "#to check messages in Kafka, go to https://lsst-kafka-0-nts-efd.ncsa.illinois.edu/\n",
+    "#we cannot get time series data from DDS. We have to query the EFD"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dfe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### check the behavior when LUT is disabled."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "lutMode = await hex.evt_compensationMode.aget(timeout=10)\n",
+    "print(\"compsensation mode enabled?\",lutMode.enabled)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos0 = await hex.tel_application.aget(timeout=10.)\n",
+    "print(pos0.MTHexapodID)\n",
+    "for i in range(6):\n",
+    "    print('%10.2f '%pos0.position[i], end= ' ')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you want to observe the motions in chronograf, consider using \"AND MTHexapodID==1\" to filter out telmetry from the other hexapod"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for step in range(5,-1,-1):\n",
+    "    #according to XML, units are micron and degree\n",
+    "    await hex.cmd_move.set_start(x=0,y=0,z=100*step, u=0,v=0,w=0,sync=True)\n",
+    "    await asyncio.sleep(0.1)\n",
+    "    state = await hex.evt_inPosition.aget(timeout=10)\n",
```

Comment:
I *strongly* suggest using `next` instead of `aget` (and skipping the sleep). `aget` is not safe in a loop because it can repeatedly return the same value if no new data is seen.

In fact this loop includes a strange mix of aget and next.

Also if a telemetry topic is output frequently, as is tel_application, then you should read it with next(flush=False) to make sure you are getting a new value.

Also I suggest waiting until you are in position before issuing the next move command.

I suggest something like the following:
```
hex.evt_inPosition.flush()
for step in range(5,-1,-1):
    await hex.cmd_move.set_start(x=0,y=0,z=100*step, u=0,v=0,w=0,sync=True)
    while True:
       state = await hex.evt_inPosition.next(flush=False, timeout=10)
       print("Hexapod in position?", state.inPosition)
       if state.inPosition:
           break
   pos = await hex.tel_application.next(flush=False, timeout=10)
   print(" ",join(f"{pos:10.2f}" for pos in pos.position)")
```